patches/roslyn-analyzers/0003-Use-individual-Microsoft.CodeAnalysis-packages.patch

```diff
@@ -0,0 +1,70 @@
+From 05ec9e1cf92b4e28011d1ba786d40acb5e5f51cb Mon Sep 17 00:00:00 2001
+From: Omair Majid <omajid@redhat.com>
+Date: Fri, 6 Nov 2020 18:18:48 -0500
+Subject: [PATCH 3/5] Use individual Microsoft.CodeAnalysis* packages
+
+The superset package Microsoft.CodeAnalysis is not built in
+source-build, so lets use the individual packages.
+---
+ eng/Versions.props                                     | 10 +++++++---
+ .../GenerateDocumentationAndConfigFiles.csproj         |  5 ++++-
+ .../Workspaces/Workspaces.Utilities.projitems          |  7 +++++--
+ 3 files changed, 16 insertions(+), 6 deletions(-)
+
+diff --git a/eng/Versions.props b/eng/Versions.props
+index ced06221d..da9b8d60a 100644
+--- a/eng/Versions.props
++++ b/eng/Versions.props
+@@ -24,10 +24,14 @@
+     <!-- Use the correct compiler version -->
+     <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
+     <!-- Roslyn -->
+-    <MicrosoftCodeAnalysisVersion>3.3.1</MicrosoftCodeAnalysisVersion>
+-    <MicrosoftCodeAnalysisCommonVersion>3.3.1</MicrosoftCodeAnalysisCommonVersion>
++    <MicrosoftCodeAnalysisVersion>3.8.0</MicrosoftCodeAnalysisVersion>
++    <MicrosoftCodeAnalysisCommonVersion>3.8.0</MicrosoftCodeAnalysisCommonVersion>
++    <MicrosoftCodeAnalysisWorkspacesCommonVersion>3.8.0</MicrosoftCodeAnalysisWorkspacesCommonVersion>
++    <MicrosoftCodeAnalysisCSharpVersion>3.8.0</MicrosoftCodeAnalysisCSharpVersion>
++    <MicrosoftCodeAnalysisVisualBasicWorkspacesVersion>3.8.0</MicrosoftCodeAnalysisVisualBasicWorkspacesVersion>
++    <MicrosoftCodeAnalysisVisualBasicVersion>3.8.0</MicrosoftCodeAnalysisVisualBasicVersion>
+     <MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion>$(MicrosoftCodeAnalysisVersion)</MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion>
+-    <MicrosoftCodeAnalysisForShippedApisVersion>3.7.0</MicrosoftCodeAnalysisForShippedApisVersion>
++    <MicrosoftCodeAnalysisForShippedApisVersion>$(MicrosoftCodeAnalysisVersion)</MicrosoftCodeAnalysisForShippedApisVersion>
```

Comment:
Heads up that this pattern *can* cause an issue if we need it to get updated by the PVP. Eval goes like:

1. `<MicrosoftCodeAnalysisVersion>3.8.0</>` -> set to 3.8.0
2. `<MicrosoftCodeAnalysisForShippedApisVersion>$(MicrosoftCodeAnalysisVersion)</>` -> set to 3.8.0 (fully evaluates now)
3. PVP overrides `MicrosoftCodeAnalysisVersion` based on `Microsoft.CodeAnalysis.*.nupkg`, say 3.8.1.
4. `MicrosoftCodeAnalysisForShippedApisVersion` is now "stranded" at 3.8.0

This can be fixed by adding one of these to make the PVP override both properties:

https://github.com/dotnet/source-build/blob/055775f3c4fd1120aeb69c404e76ce33a48cec05/Directory.Build.props#L255-L256

Depending on what this is used for, might be worth adding an `ExtraPackageVersionPropsPackageInfo` entry now. Shouldn't have any effect now, but might help us out with 5.0.101 or 102 if Roslyn gets updates.

Reply:
Thanks! Fixed now.